### DISCUSSION TITLE
Can't store embeddings in Milvus DB

### DISCUSSION THREAD

**Discussion by MW3r3** [milvus-io/milvus#27243] [Q&A and General discussion] [answered]
Im trying to make a chatbot with long term memory using embeddings however I cant manage to store embedding I get from OpenAI here is the functions I use
This is the schema I use
```
if not utility.has_collection("practyai_vdb"):
    dim = 1536

    fields = [
        FieldSchema(name="pk", dtype=DataType.INT64, is_primary=True, auto_id=True),
        FieldSchema(name="embeddings", dtype=DataType.FLOAT_VECTOR, dim=dim)
    ]
    schema = CollectionSchema(fields, "This is a vector database for storing user and bot messages for PractyAI")

    vdb = Collection("practyai_vdb", schema, consistency_level="Strong")
    print("vdb successfully created")

else:
    vdb = Collection("practyai_vdb")
```
this the function I use to store embeddings I changed it multiple time and couldn't find the first version however it was wrong anyways
```
def insert_to_vdb(embedding):
    global vdb
    entities = [
        [],
        embedding,
    ]
    insert_result = vdb.insert(entities)
    print("successfully inserted into vector database")
    return insert_result
```
here is how I get embeddings from openai
```
def get_embedding_from_openai(message):
    response = openai.Embedding.create(
        engine="text-embedding-ada-002",
        input=message
    )
    print("successfully fetched embedding from OpenAI")
    embedding_list = response['data'][0]['embedding']
    return embedding_list
```
the code is giving an error saying expecting FLOAT_VECTOR given DOUBLE I cant manage to fix it
any help is appreciated.

**Comment by xiaofan-luan** [comment]
what is the error you get?
openAI embeddings is Float vector but not double

**Reply by MW3r3** [reply]
The error i get was something like expected float_Vector given Double i managed to solve it by for looping open ai's vector output and converting each to float from double

**Comment by MW3r3** [comment, marked as answer]
i manage to solve the problem by changing the values in side of embeddings output from openai
```def get_embedding_from_openai(message):
    response = openai.Embedding.create(
        engine="text-embedding-ada-002",
        input=message
    )
    print("successfully fetched embedding from OpenAI")
    embedding_list = response['data'][0]['embedding']
    embedding_float_list = []
    for double_value in embedding_list:
        float_value = float(double_value)  # Convert each double value to float
        embedding_float_list.append(float_value)  # Append the float value to the new list
    return embedding_float_list```

**Comment by xiaofan-luan** [comment]
> i manage to solve the problem by changing the values in side of embeddings output from openai
> 
> ```
>     response = openai.Embedding.create(
>         engine="text-embedding-ada-002",
>         input=message
>     )
>     print("successfully fetched embedding from OpenAI")
>     embedding_list = response['data'][0]['embedding']
>     embedding_float_list = []
>     for double_value in embedding_list:
>         float_value = float(double_value)  # Convert each double value to float
>         embedding_float_list.append(float_value)  # Append the float value to the new list
>     return embedding_float_list```
> ```



> i manage to solve the problem by changing the values in side of embeddings output from openai
> 
> ```
>     response = openai.Embedding.create(
>         engine="text-embedding-ada-002",
>         input=message
>     )
>     print("successfully fetched embedding from OpenAI")
>     embedding_list = response['data'][0]['embedding']
>     embedding_float_list = []
>     for double_value in embedding_list:
>         float_value = float(double_value)  # Convert each double value to float
>         embedding_float_list.append(float_value)  # Append the float value to the new list
>     return embedding_float_list```
> ```


Using the Python library provided by OpenAI, embeddings requests return floats with up to 18 decimal places of precision. Using other methods (curl, hyper/reqwest) only returns floats with half the level of precision.

thanks for the finding

**Reply by xiaofan-luan** [reply]
referene https://community.openai.com/t/discrepancy-in-embeddings-precision/78666
